### ISSUE TITLE
Turn instruction doesn't update for an upcoming turn till the very end after a reroute

### ISSUE THREAD

**Issue by souperneon** [mapzen/eraser-map#227]
When an exit/turn is right after a reroute has happened the instruction changes only a few feet before

Start point: Austerlitz, NY
End point: 31-19, 33rd St, Astoria NY 11106

This happened to me multiple times in the route…
Original route asked me to turn left on 30th av and right on 31st st (after getting off the triborough bridge). But I kept driving on 30th street and turned left on broadway. The route had just updated from a reroute and didn't show me the instruction to turn left on 33rd St until I was AT the turn (see screenshot)

![screenshot_2016-01-03-14-25-16](https://cloud.githubusercontent.com/assets/6125224/12128517/d56eed6c-b3cb-11e5-9e74-df32286d6008.png)

**Comment by dgearhart** [comment]
* **Device name:** samsung Model: SM-G900V
* **Android Version:** 5.0
* **App build number:** master-674
* **What did you expected to happen?** When I take a different route between by origin and destination that the app recalculates and stays in sync with my current location on the map and the narrative call-outs.
* **What happened instead?** The app recalculated however it did not stay in sync with my current location. The maneuver information displayed at the top of the screen remained on the previous maneuver even though the map position was updating withmyu current position. Please see screen shots. Also, the next verbal narrative call-out did not happen until right at the turn - which the user would have missed if the route was not known. It seems that the maneuver display and call-outs are behind until the actual transition point or the screen display toggles between portrait and landscape.
* **Steps to reproduce:** Deviate from the the initial route.
* **Attach a screenshot**
User is already on Lititz Pike but previous maneuver is still displayed. More importantly, the next maneuver call-out comes too late since it happens right at the transition point.
![screenshot_2016-01-19-08-00-27](https://cloud.githubusercontent.com/assets/7515853/12423391/3b774438-be99-11e5-91ca-5e1493342cb6.png)

Again, user is already on North Plum Street but previous maneuver is still displayed on both screen shots below:
![screenshot_2016-01-19-08-02-30](https://cloud.githubusercontent.com/assets/7515853/12423696/b4b109f0-be9a-11e5-9dd5-31bfa5262cee.png)
![screenshot_2016-01-18-07-49-26](https://cloud.githubusercontent.com/assets/7515853/12423707/c45bce1c-be9a-11e5-9f03-6991af0b750d.png)

* **Routing from origin:** {"lat":40.06912263645471,"lon":-76.31102886982262,"type":"break"}
* **Routing through:** {"lat":40.05664983732192,"lon":-76.30090529099107,"type":"through"}
* **Routing through:** {"lat":40.05411229981102,"lon":-76.29955345764756,"type":"through"}
* **Routing to destination:** {"lat":40.04258954139421,"lon":-76.29926377907395,"type":"break"}

**Comment by mjcunningham** [comment]
@ecgreb I'm moving this into beta.

**Comment by souperneon** [comment]
This is almost exactly what I reported as well. After reroute, the instructions fall behind and don't give the same alerts for upcoming turns. @ecgreb said it was probably because of the threshold? like if I missed an exit and the reroute has me making a turn fairly immediately after, the verbal instruction won't happen at all because it has crossed the threshold of happening a mile before the maneuver. Not sure if I made any sense here ;)
